File: backend/app/api/models/note.py
from pydantic import BaseModel, Field
from datetime import datetime

class NoteModel(BaseModel):
    id: int = Field(...)
    dateStart: datetime = Field(...)
    dateEnd: datetime = Field(...)
    text: str = Field(..., min_length=0, max_length=256)
    color: str = Field(..., min_length=6, max_length=6)

File: frontend/src/structures/Note.ts
export interface NoteModel {
  id: number;
  dateStart: Date;
  dateEnd: Date;
  text: string;
  color: string;
}

export const defaultNoteModel: NoteModel = {
  id: 0,
  dateStart: new Date(),
  dateEnd: new Date(),
  text: "",
  color: "ffffff",
};
